app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,57 @@
+class PostsController < ApplicationController
+
+  
+ 
+  #to render a particular post(s).
+  def show
+    @post = Post.find(params[:id])
+  end
+  
+  #to return a view with an HTML form to create a new post
+  def new
+   @post = Post.new 
+  end
+
+  #actually do the creating of said post
+  def create
+    @post = Post.new(post_params)
+    if @post.save
+      redirect_to root_path
+    else
+      render 'new'
+    end
+  end
+
+  private
+    def post_params
+      params.require(:post).permit(:title, :url)
+    end
```

Comment:
FYI: you don't need to do additional indenting in the `private` section of a class.